test/multitaper.jl

```diff
@@ -0,0 +1,156 @@
+    @test several_signals_coherences[2, 1] ≈ phase_shift_coherence
+    @test several_signals_coherences[3, 1] ≈ different_signal_coherence
+
+    # test in-place gets the same result
+    config = MTCoherenceConfig{eltype(several_signals)}(size(several_signals)...; fs=fs, freq_range = (10, 15))
+    out = allocate_output(config)
+    coh2 = mt_coherence!(out, several_signals, config)
+    @test several_signals_coherences ≈ coh2
+end
+
+@testset "`mt_coherence` reference test" begin
+    fs = 1000.0
+    n_samples = 1024
+    t = (0:1023) ./ fs
+    sin_1 = sin.(2 * π * 12.0 * t)  # 12 Hz sinusoid signal
+    sin_2 = sin.(2 * π * 12.0 * t .+ π)
+    noise = vec(readdlm(joinpath(@__DIR__, "data", "noise.txt"))) # generated by `rand(1024) * 2 .- 1`
+    more_noisy = Array{Float64}(undef, 1, 2, n_samples)
+    more_noisy[1, 1, :] = sin_1
+    more_noisy[1, 2, :] .= sin_1 .+ 3 * noise
+
+    ## to generate the reference result:
+    # using PyMNE
+    # mne_coherence_matrix, _ = PyMNE.connectivity.spectral_connectivity(more_noisy, method="coh", sfreq=fs,mode="multitaper",fmin=10,fmax=15,verbose=false)
+    # coh = dropdims(mean(mne_coherence_matrix; dims=3); dims=3)[2, 1]
+    coh = 0.982356762670818
+    result = mt_coherence(dropdims(more_noisy;dims=1); fs=fs, freq_range = (10,15))
+    @test result[2, 1] ≈ coh
+end
+
+@testset "`mt_cross_spectral`" begin
+    fs = 1000.0
+    n_samples = 1024
+    t = (0:1023) ./ fs
+    data = Array{Float64, 3}(undef, 1, 2, n_samples)
+    data[1,1,:] = sin.(2 * π * 12.0 * t) # 12 Hz sinusoid signal
+    data[1,2,:] = sin.(2 * π * 12.0 * t .+ π)
+    ## Script to generate reference data:
+    # using PyMNE, DelimitedFiles
+    # result = PyMNE.time_frequency.csd_array_multitaper(data, fs, n_fft = nextpow(2, n_samples), low_bias=true, adaptive=false)
+    # csd_array_multitaper_frequencies = result.frequencies
+    # csd_array_multitaper_values = reduce((x,y) -> cat(x,y; dims=3), [ r.get_data() for r in result])
+    # writedlm("data/csd_array_multitaper_frequencies.txt", csd_array_multitaper_frequencies)
+    # writedlm("data/csd_array_multitaper_values_re.txt", real(csd_array_multitaper_values))
+    # writedlm("data/csd_array_multitaper_values_im.txt", imag(csd_array_multitaper_values))
+    csd_array_multitaper_frequencies = readdlm(joinpath(@__DIR__, "data", "csd_array_multitaper_frequencies.txt"))
+    csd_array_multitaper_values_re = reshape(readdlm(joinpath(@__DIR__, "data", "csd_array_multitaper_values_re.txt")), (2,2,512))
+    csd_array_multitaper_values_im = reshape(readdlm(joinpath(@__DIR__, "data", "csd_array_multitaper_values_im.txt")), (2,2,512))
+    csd_array_multitaper_values = csd_array_multitaper_values_re + im*csd_array_multitaper_values_im
+
+    signal = dropdims(data; dims=1)
+    result = mt_cross_spectral(signal; fs=fs)
+    @test freq(result)[2:end] ≈ csd_array_multitaper_frequencies
+    @test result.values[:,:,2:end] ≈ csd_array_multitaper_values rtol=1e-3
```

Comment:
only matches up to `rtol=1e-3`; I think that means it doesn't actually match to numerical precision, but I spent a long time trying to see what was different and failed to find anything more

Reply:
seems to work now!